presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/aggregation/TestConvexHullAgg.java

```diff
@@ -0,0 +1,380 @@
+                },
+                {
+                        "square with adjacent line",
+                        "POLYGON ((0 5, 1 1, 3 1, 5 5, 0 5))",
+                        new String[] {"POLYGON ((1 1, 3 1, 3 3, 1 3, 1 1))", "LINESTRING (0 5, 5 5)"},
+                },
+                {
+                        "square with adjacent point",
+                        "POLYGON ((5 2, 3 3, 1 3, 1 1, 3 1, 5 2))",
+                        new String[] {"POLYGON ((1 1, 3 1, 3 3, 1 3, 1 1))", "POINT (5 2)"},
+                },
+        };
+    }
+
+    private InternalAggregationFunction function;
+
+    @BeforeClass
+    public void registerFunctions()
+    {
+        GeoPlugin plugin = new GeoPlugin();
+        for (Type type : plugin.getTypes()) {
+            functionAssertions.getTypeRegistry().addType(type);
+        }
+        functionAssertions.getMetadata().addFunctions(extractFunctions(plugin.getFunctions()));
+        function = functionAssertions.getMetadata().getFunctionRegistry().getAggregateFunctionImplementation(new
+                Signature("convex_hull_agg",
+                FunctionKind.AGGREGATE,
+                TypeSignature.parseTypeSignature(GeometryType.GEOMETRY_TYPE_NAME),
+                TypeSignature.parseTypeSignature(GeometryType.GEOMETRY_TYPE_NAME)));
+    }
+
+    @Test(dataProvider = "point")
+    public void testPoint(String testDescription, String expectedWkt, String... wkts)
+    {
+        assertAggregatedGeometries(testDescription, expectedWkt, wkts);
+    }
+
+    @Test(dataProvider = "linestring")
+    public void testLineString(String testDescription, String expectedWkt, String... wkts)
+    {
+        assertAggregatedGeometries(testDescription, expectedWkt, wkts);
+    }
+
+    @Test(dataProvider = "polygon")
+    public void testPolygon(String testDescription, String expectedWkt, String... wkts)
+    {
+        assertAggregatedGeometries(testDescription, expectedWkt, wkts);
+    }
+
+    @Test(dataProvider = "multipoint")
+    public void testMultipoint(String testDescription, String expectedWkt, String... wkt)
+    {
+        assertAggregatedGeometries(testDescription, expectedWkt, wkt);
+    }
+
+    @Test(dataProvider = "multilinestring")
+    public void testMultilinestring(String testDescription, String expectedWkt, String... wkt)
+    {
+        assertAggregatedGeometries(testDescription, expectedWkt, wkt);
+    }
+
+    @Test(dataProvider = "multipolygon")
+    public void testMultipolygon(String testDescription, String expectedWkt, String... wkt)
+    {
+        assertAggregatedGeometries(testDescription, expectedWkt, wkt);
+    }
+
+    @Test(dataProvider = "geometrycollection")
+    public void testGeometrycollection(String testDescription, String expectedWkt, String... wkt)
+    {
+        assertAggregatedGeometries(testDescription, expectedWkt, wkt);
+    }
+
+    private void assertAggregatedGeometries(String testDescription, String expectedWkt, String... wkts)
```

Comment:
To reduce copy-paste, consider moving `assertAggregatedGeometries` and `registerFunctions` methods to an abstract base class that's used by this test and a test for `geometry_union_agg` function.